src/etc/inc/interfaces.inc

```diff
@@ -1342,7 +1342,7 @@ function interface_vxlan_configure(&$vxlan, $vxlankey = "") {
 		log_error(gettext("could not bring parentif up -- variable not defined -- interface_vxlan_configure()"));
 	}
 
-	if (!platform_booting() && !(empty($vxlan['vxlanif']))) {
+	if (platform_booting() || !(empty($vxlan['vxlanif']))) {
```

Comment:
I didn't understand the reason for this change.  Could you please elaborate a bit?

Reply:
I confess the exact logic here eludes me! I tested it that was as that's how gif and gre interfaces are configured and that seems to work.
With the original logic the interface is created twice at boot with the same vxlanid and subsequent vxlans use sequential numbers rather that that defined in the config. Like:

`Configuring WAN interface...
vtnet0: link state changed to UP
vxlan0: Ethernet address: 58:9c:fc:10:cb:30
vxlan0: link state changed to UP
done.
Configuring VXLAN0 interface...done.
vxlan1: Ethernet address: 58:9c:fc:10:ff:9f
vxlan1: network identifier 12345 already exists in this socket
vxlan1: network identifier 12345 already exists in this socket
Configuring LAN interface...
vtnet1: link state changed to UP
done.
vxlan2: Ethernet address: 58:9c:fc:10:ff:99
vxlan2: link state changed to UP`

Reply:
Seems similar to what we were seeing with VTI ipsec interfaces, might need similar logic change to what was done there. Basically only destroy it if the interface already exists. And maybe also test if it exists before creating.

Reply:
@stephenw10 could you try to replace this change by:
```
if (does_interface_exist($vxlan['vxlanif'])) {
```
And see if it has the same effect?

Reply:
It behaves differently but still incorrectly. The first interface is created at boot as expected though it is still brought UP then DOWN then back UP again.
The second interface is created as vxlan1 though the config is for vxlan3 resulting in a spurious interface in the GUI Interfaces > Assign.
```Configuring IPsec VTI interfaces...done.
Configuring WAN interface...
vtnet0: link state changed to UP
vxlan0: Ethernet address: 58:9c:fc:10:cb:30
vxlan0: link state changed to UP
done.
Configuring VXLAN0 interface...done.
vxlan0: link state changed to DOWN
vxlan0: Ethernet address: 58:9c:fc:10:cb:30
vxlan0: link state changed to UP
Configuring LAN interface...
vtnet1: link state changed to UP
done.
vxlan1: Ethernet address: 58:9c:fc:10:ff:9f
vxlan1: link state changed to UP```

From config:
```	<vxlans>
		<vxlan>
			<if>wan</if>
			<ipaddr></ipaddr>
			<remote-addr>172.21.16.187</remote-addr>
			<vxlanid>12345</vxlanid>
			<vxlanlocalport></vxlanlocalport>
			<vxlanremoteport></vxlanremoteport>
			<vxlanttl></vxlanttl>
			<vxlanlearn></vxlanlearn>
			<descr><![CDATA[Test VXLAN]]></descr>
			<vxlanif>vxlan0</vxlanif>
		</vxlan>
		<vxlan>
			<if>lan</if>
			<ipaddr></ipaddr>
			<remote-addr>192.168.180.56</remote-addr>
			<vxlanid>54322</vxlanid>
			<vxlanlocalport></vxlanlocalport>
			<vxlanremoteport></vxlanremoteport>
			<vxlanttl></vxlanttl>
			<vxlanlearn></vxlanlearn>
			<descr><![CDATA[Test VXLAN 3]]></descr>
			<vxlanif>vxlan3</vxlanif>
		</vxlan>
	</vxlans>```